src/hpv_query.py

```diff
@@ -83,63 +84,55 @@ def process_one_file(file_path: str, text_key: str = "content") -> Iterable[dict
 def process_all_files(
     file_paths: list,
     save_file: pathlib.Path,
-    log_file: pathlib.Path,
     text_key="content",
 ):
     """
-    Process all files with an option to overwrite existing results and automatically reprocess the last file in the log if it exists.
-
-    Reprocessing the last file ensures that no data is complete if the process was interrupted mid-file.
+    Process all files and save the results in a jsonl file.
 
     args:
         file_paths: list
         save_file: pathlib.Path
-        log_file: pathlib.Path
         text_key: str
     """
-    for file in [save_file, log_file]:
-        if (
-            file.exists()
-        ):  # since we are appending to the logfile + savefile, we want to remove it if we re-run the script (to avoid duplicates)
-            file.unlink()
+    if save_file.exists() and save_file.suffix == ".jsonl": 
```

Comment:
I'm not sure I understand the rationale behind checking if the suffix is `.jsonl`.
What if the user specifies the file as `hpv_entries.ndjson`? Should the file not be deleted then?
Also, since you are replacing the file, it might be simply worth to not delete it at all and use:
```python
save_file.open("w")
```
instead, as this will delete previous file contents automatically.

Reply:
`jsonl` was to constrain the user, but I can see that it makes no sense. The save_file.open("w") is a good shout!